packages/edge-gateway/src/gateway.js

```diff
@@ -525,3 +531,22 @@ function getDurableRequestUrl(request, route, data) {
     body: data && JSON.stringify(data),
   })
 }
+
+/**
+ * Transforms race response with custom headers.
+ * Content-Security-Policy header specified to only allow requests within same origin.
+ *
+ * @param {Response} response
+ */
+function getTransformedResponseWithCustomHeaders(response) {
+  const clonedResponse = new Response(response.body, {
+    headers: response.headers,
+  })
```

Comment:
To retain the `status`:

```suggestion
  const clonedResponse = new Response(response.body, response)
```